File: trunk/feature_tests/TextInput.rb
require '../gosu'

class Test < Gosu::Window
  def initialize
    super(300, 200, false, 20)
    @font = Gosu::Font.new(self, "Arial", 20)
    self.text_input = Gosu::TextInput.new
  end
  
  def draw
    sel_x = 10 + @font.text_width(text_input.text[0...text_input.selection_start])
    pos_x = 10 + @font.text_width(text_input.text[0...text_input.caret_pos])
    
    draw_quad(sel_x, 100, 0xcc0000ff, pos_x, 100, 0xcc0000ff,
              sel_x, 120, 0xcc0000ff, pos_x, 120, 0xcc0000ff)
              
    draw_line(pos_x, 100, 0xffffffff, pos_x, 120, 0xffffffff, 0)    
    
    @font.draw(text_input.text, 10, 100, 0)
  end
end

Test.new.show

File: trunk/examples/TextInput.cpp
#include <Gosu/Gosu.hpp>
#include <boost/scoped_ptr.hpp>

class TextInputWindow : public Gosu::Window
{
    boost::scoped_ptr<Gosu::Font> font;
    boost::scoped_ptr<Gosu::TextInput> textInput;

public:
    TextInputWindow()
    : Gosu::Window(300, 200, false)
    {
        setCaption(L"Text Input Example");
        font.reset(new Gosu::Font(graphics(), Gosu::defaultFontName(), 20));
        textInput.reset(new Gosu::TextInput());
        input().setTextInput(textInput.get());
    }
    
    void draw()
    {
        std::wstring text = textInput->text();
        double selX = 10 + font->textWidth(text.substr(0, textInput->selectionStart()));
        double posX = 10 + font->textWidth(text.substr(0, textInput->caretPos()));
        
        graphics().drawQuad(selX, 100, 0xcc0000ff, posX, 100, 0xcc0000ff,
                            selX, 120, 0xcc0000ff, posX, 120, 0xcc0000ff, 0);
                 
        graphics().drawLine(posX, 100, 0xffffffff, posX, 120, 0xffffffff, 0);
        
        font->draw(text, 10, 100, 0);
    }
};

int main(int argc, char* argv[])
{
    TextInputWindow win;
    win.show();
    return 0;
}
